huaweicloud/services/nat/resource_huaweicloud_nat_gateway.go

```diff
@@ -0,0 +1,293 @@
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The ID of the VPC to which the NAT gateway belongs.",
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Description: "The network ID of the downstream interface (the next hop of the DVR) " +
+					"of the NAT gateway.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description fo the NAT gateway.",
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The enterprise project ID of the NAT gateway.",
+			},
+			"tags": common.TagsSchema(),
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The current status of the NAT gateway.",
+			},
+		},
+	}
+}
+
+func gatewayStateRefreshFunc(client *golangsdk.ServiceClient, gatewayId string,
+	targets []string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := gateways.Get(client, gatewayId)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				return resp, "COMPLETED", nil
+			}
+			return resp, "", err
+		}
+
+		if utils.StrSliceContains([]string{"INACTIVE"}, resp.Status) {
+			return resp, "", fmt.Errorf("unexpect status (%s)", resp.Status)
+		}
+		if utils.StrSliceContains(targets, resp.Status) {
+			return resp, "COMPLETED", nil
+		}
+		return resp, "PENDING", nil
+	}
+}
+
+func resourceGatewayCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NatGatewayClient(region)
+	if err != nil {
+		return diag.Errorf("error creating NAT v2 client: %s", err)
+	}
+
+	opts := gateways.CreateOpts{
+		Name:                d.Get("name").(string),
+		VpcId:               d.Get("vpc_id").(string),
+		InternalNetworkId:   d.Get("subnet_id").(string),
+		Spec:                d.Get("spec").(string),
+		Description:         d.Get("description").(string),
+		EnterpriseProjectId: common.GetEnterpriseProjectID(d, cfg),
+	}
+	resp, err := gateways.Create(client, opts)
+	if err != nil {
+		return diag.Errorf("error creating NAT gateway: %s", err)
+	}
+	d.SetId(resp.ID)
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      gatewayStateRefreshFunc(client, d.Id(), []string{"ACTIVE"}),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        5 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if gatewayTags := d.Get("tags").(map[string]interface{}); len(gatewayTags) > 0 {
```

Comment:
I think using GetOk is a more common way

Reply:
done.